src/ImageSharp/Formats/Webp/Lossy/Vp8BandProbas.cs

```diff
@@ -20,6 +22,13 @@ public Vp8BandProbas()
         }
     }
 
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Vp8BandProbas"/> class.
+    /// Only used for unit tests.
+    /// </summary>
+    [JsonConstructor]
```

Comment:
We should probably use a [custom JSON converter](https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/converters-how-to?pivots=dotnet-7-0) in the tests rather than adding attributes to the source.

Reply:
Sorry @brianpopow I've given you the runaround here. I didn't realize you'd introduced the constructors.

Let's go with your original plan but wrap the namespace imports and constructor code in the following conditional.

```c#
#if SIXLABORS_TESTING || SIXLABORS_TESTING_PREVIEW
```

Reply:
>  Let's go with your original plan but wrap the namespace imports and constructor code in the following conditional.
> #if SIXLABORS_TESTING || SIXLABORS_TESTING_PREVIEW

I dont think that's a good idea. If someone does  a fresh clone of the repo and tries to run the tests, those webp tests will fail without any obvious reason. I dont think test should behave different when a environment is set. Unfortunately I dont have a better idea so far.

Reply:
Can we avoid the json and populate the test instance with equivalent data generated in a class using string builder or similar?

Reply:
> Can we avoid the json and populate the test instance with equivalent data generated in a class using string builder or similar?

Usually I would create a test instance in the unittest and populate it with data, but for `Vp8Residual` that does not seem feasible. There are so many nested arrays, that I dont know how this can be populated manually. Thats why I have chosen to go for json serialization.

Also I dont understand your suggestion with the string builder.

I now tried to use the binary serialization and this was looking promising. We would not need the additional constructors.
But now I noticed its marked as obsolete 😢 

See: https://learn.microsoft.com/de-de/dotnet/standard/serialization/binaryformatter-security-guide

Reply:
Changed serialization now to use BinaryWriter/BinaryReader, according to https://learn.microsoft.com/de-de/dotnet/standard/serialization/binaryformatter-security-guide this should be fine.

No additional constructors required now.

Reply:
Ah that’s fantastic! We’ll done finding a solution!